extensions/barrage/src/main/java/io/deephaven/extensions/barrage/chunk/array/ShortArrayExpansionKernel.java

```diff
@@ -16,61 +16,111 @@
+public class ShortArrayExpansionKernel implements ArrayExpansionKernel<short[]> {
+    public static final ShortArrayExpansionKernel INSTANCE = new ShortArrayExpansionKernel();
+
+    private static final String DEBUG_NAME = "ShortArrayExpansionKernel";
+    private static final short[] ZERO_LEN_ARRAY = new short[0];
 
     @Override
-    public <T, A extends Any> WritableChunk<A> expand(final ObjectChunk<T, A> source,
-            final WritableIntChunk<ChunkPositions> perElementLengthDest) {
+    public <A extends Any> WritableChunk<A> expand(
+            @NotNull final ObjectChunk<short[], A> source,
+            final int fixedSizeLength,
+            @Nullable final WritableIntChunk<ChunkPositions> offsetsDest) {
         if (source.size() == 0) {
-            perElementLengthDest.setSize(0);
+            if (offsetsDest != null) {
+                offsetsDest.setSize(0);
+            }
             return WritableShortChunk.makeWritableChunk(0);
         }
 
-        final ObjectChunk<short[], A> typedSource = source.asObjectChunk();
-
         long totalSize = 0;
-        for (int i = 0; i < typedSource.size(); ++i) {
-            final short[] row = typedSource.get(i);
-            totalSize += row == null ? 0 : row.length;
+        for (int ii = 0; ii < source.size(); ++ii) {
+            final int rowLen;
+            if (fixedSizeLength != 0) {
+                rowLen = Math.abs(fixedSizeLength);
+            } else {
+                final short[] row = source.get(ii);
+                rowLen = row == null ? 0 : row.length;
+            }
+            totalSize += rowLen;
         }
         final WritableShortChunk<A> result = WritableShortChunk.makeWritableChunk(
-                LongSizedDataStructure.intSize("ExpansionKernel", totalSize));
+                LongSizedDataStructure.intSize(DEBUG_NAME, totalSize));
 
         int lenWritten = 0;
-        perElementLengthDest.setSize(source.size() + 1);
-        for (int i = 0; i < typedSource.size(); ++i) {
-            final short[] row = typedSource.get(i);
-            perElementLengthDest.set(i, lenWritten);
-            if (row == null) {
-                continue;
+        if (offsetsDest != null) {
+            offsetsDest.setSize(source.size() + 1);
+        }
+        for (int ii = 0; ii < source.size(); ++ii) {
+            final short[] row = source.get(ii);
+            if (offsetsDest != null) {
+                offsetsDest.set(ii, lenWritten);
+            }
+            int written = 0;
+            if (row != null) {
+                int offset = 0;
+                if (fixedSizeLength != 0) {
+                    // limit length to fixedSizeLength
+                    written = Math.min(row.length, Math.abs(fixedSizeLength));
+                    if (fixedSizeLength < 0 && written < row.length) {
+                        // read from the end of the array when fixedSizeLength is negative
```

Comment:
Right now if you have a long list as shown to the UI, we preview only the beginning of it. It could be a nice feature to preview from the end. That said, Colin mentioned that there is some plotting feature that would benefit from seeing only the end of arrays and that is the purpose for being able to tail the preview.